src/Association/Embedded.php

```diff
@@ -64,48 +66,76 @@ public function __construct($alias, $options = [])
     {
         $this->alias = $alias;
         $properties = [
-            'entityClass',
-            'property',
-            'indexClass',
+            'entityClass' => 'setEntityClass',
+            'property' => 'setProperty',
+            'indexClass' => 'setIndexClass',
         ];
         $options += [
             'entityClass' => $alias,
         ];
-        foreach ($properties as $prop) {
+        foreach ($properties as $prop => $method) {
             if (isset($options[$prop])) {
-                $this->{$prop}($options[$prop]);
+                $this->{$method}($options[$prop]);
             }
         }
     }
 
+    /**
+     * Get the property this embed is attached to.
+     *
+     * @return string The property name.
+     */
+    public function getProperty()
```

Comment:
This method hasn't been split into a get/set pair in the 3.x branch yet.

Reply:
I'm focusing on 2.x because I need that plugin with CakePHP 3.x.

But I can send other PR with some of that changes to master if you want.

Reply:
That would be great.